include/cpp-statsd-client/UDPSender.hpp

```diff
@@ -166,30 +169,39 @@ inline UDPSender::~UDPSender() {
     close(m_socket);
 }
 
-inline void UDPSender::send(const std::string& message) noexcept {
+inline void UDPSender::send(const std::string::const_iterator& begin, const std::string::const_iterator& end) noexcept {
     m_errorMessage.clear();
     // If batching is on, accumulate messages in the queue
     if (m_batching) {
         std::unique_lock<std::mutex> batchingLock(m_batchingMutex);
+        // Either we don't have a place to batch our message or we exceeded the batch size, so make a new batch
         if (m_batchingMessageQueue.empty() || m_batchingMessageQueue.back().length() > m_batchsize) {
-            m_batchingMessageQueue.push_back(message);
-        } else {
-            m_batchingMessageQueue.rbegin()->append("\n").append(message);
+            m_batchingMessageQueue.emplace_back();
+            m_batchingMessageQueue.back().reserve(m_batchsize + 256);
```

Comment:
when we need a new batch started, we now reserve enough bytes that we will not need to reallocate it again.

Reply:
We could make this function slightly clearer with something like

```cpp
inline void UDPSender::send(const std::string& message) noexcept {
    m_errorMessage.clear();
    // If batching is on, accumulate messages in the queue
    if (m_batching) {
        queueMessage(message);
        return;
    }

    sendToDaemon(message);
}

inline void UDPSender::queueMessage(const std::string& message) noexcept {
    std::unique_lock<std::mutex> batchingLock(m_batchingMutex);
    // Either we don't have a place to batch our message or we exceeded the batch size, so make a new batch
    if (m_batchingMessageQueue.empty() || m_batchingMessageQueue.back().length() > m_batchsize) {
        m_batchingMessageQueue.emplace_back();
        m_batchingMessageQueue.back().reserve(m_batchsize + 256);
    }  // When there is already a batch open we need a separator when its not empty
    else if (!m_batchingMessageQueue.back().empty()) {
        m_batchingMessageQueue.back().push_back('\n');
    }
    // Add the new message to the batch
    m_batchingMessageQueue.back().append(message);
}
```

Reply:
done!